spec/models/searchgov_url_spec.rb

```diff
@@ -890,6 +890,65 @@
       end
     end
 
+    context 'when searchgov url page has different canonical_url' do
+      let(:canonical_url) { 'https://www.agency.gov/canonical' }
+      let(:canon) { described_class.new(url: canonical_url) }
+      let(:html) do
+        <<~HTML
+          <!doctype html>
+          <html lang="en">
+            <head>
+              <title>Canonical Title</title>
+              <link rel="canonical" href="#{canonical_url}">
+            </head>
+            <body>
+              <p>Canonical content.</p>
+            </body>
+          </html>
+        HTML
+      end
+
+      before do
+        allow(described_class).to receive(:create!).and_call_original
+        stub_request(:get, url).
+          to_return(status: 200,
+                    body: html,
+                    headers: { content_type: 'text/html' })
+      end
+
+      it 'does not fetch the original url' do
+        fetch
+        expect(described_class).not_to have_received(:create!).with(url: url)
+      end
+
+      context 'when canonical url already exists' do
+        before do
```

Comment:
The setup for this and what's on line 941 are the same.  They aren't actually different conditions.